File: FuckLeetcode/45. 跳跃游戏 II.py
class Solution:
    def jump(self, nums) -> int:
        n = len(nums)
        dp = [float('inf')]*n
        dp[0] = 0
        for i in range(n):
            for j in range(i, min(i+nums[i]+1, n)):
                # print(j)
                dp[j] = min(dp[i]+1, dp[j])
        # print(dp)
        return dp[n-1]
s = Solution()
print(s.jump([2,3,1,1,4]))

File: FuckLeetcode/45. 跳跃游戏 II.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int* dp = new int[n];
        for (int i=0; i<n; i++) {
            // cout<<nums[i];
            dp[i] = 23333333;
        }
        // cout<<endl;

        dp[0] = 0;
        for (int i=0; i<n; i++) {
            int e_idx = min(n, i+nums[i]+1);
            for (int j=i; j<e_idx; j++) {
                dp[j] = min(dp[j], dp[i]+1);
            }
        }
        return dp[n-1];
    }
};

int main(int argc, char const *argv[]) {
    Solution s;
    std::vector<int> ya = {2,3,1,1,4};
    
    cout<<s.jump(ya);
    return 0;
}
